common/djangoapps/student/models.py

```diff
@@ -222,60 +222,19 @@ def user_by_anonymous_id(uid):
         return None
 
 
-def is_username_retired(username):
-    """
-    Checks to see if the given username has been previously retired
-    """
-    locally_hashed_usernames = user_util.get_all_retired_usernames(
-        username,
-        settings.RETIRED_USER_SALTS,
-        settings.RETIRED_USERNAME_FMT
-    )
-
-    # TODO: Revert to this after username capitalization issues detailed in
-    # PLAT-2276, PLAT-2277, PLAT-2278 are sorted out:
-    # return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
-
-    # Avoid circular import issues
-    from openedx.core.djangoapps.user_api.models import UserRetirementStatus
-
-    # Sandbox clean builds attempt to create users during migrations, before the database
-    # is stable so UserRetirementStatus may not exist yet. This workaround can also go
-    # when we are done with the username updates.
-    try:
-        return User.objects.filter(username__in=list(locally_hashed_usernames)).exists() or \
-            UserRetirementStatus.objects.filter(original_username=username).exists()
-    except ProgrammingError as exc:
-        # Check the error message to make sure it's what we expect
-        if "user_api_userretirementstatus" in text_type(exc):
-            return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
-        raise
-
-
 def username_exists_or_retired(username):
     """
     Check a username for existence -or- retirement against the User model.
     """
+    from student.models_api import is_username_retired
```

Comment:
Avoiding a circular import here.

TODO: move this into `models_api.py` as well.